quix-frontend/client/src/lib/runner/services/autocomplete/autocomplete-service.ts

```diff
@@ -46,39 +55,60 @@ export async function setupCompleters(
       query,
       position
     );
+
     const contextCompletions: Promise<AceCompletion[]> = sqlAutocompleter.getCompletionItemsFromQueryContext(
       queryContext
     );
 
     const [keywords, completions] = await Promise.all([
       keywordsCompletions,
-      contextCompletions,
+      contextCompletions
     ]);
 
-    let all =
-      queryContext.contextType === ContextType.Undefined
-        ? keywords
-        : completions;
+    const filteredCompletions: object[] = completions.filter(obj => obj.value.includes(prefix));
+    const searchInObject: Boolean = filteredCompletions.length === 0;
+    if (searchInObject) {
+      nestedColumnCompletions = sqlAutocompleter.getCompletionItemsFromQueryContextColumn(
+        queryContext
+      );
+      [nestedColumnCompletions] = await Promise.all([nestedColumnCompletions]);
+    }
 
-    if (prefix) {
-      const lowerCasedPrefix = prefix.trim().toLowerCase();
+    if (searchInObject && nestedColumnCompletions) {
+      all = queryContext.contextType === ContextType.Undefined ? keywords : nestedColumnCompletions;
+    } else {
+      all = completions;
+    }
 
```

Comment:
thought you said not to do it after convo with יבגני.
noted and will do it now.
will comment again when done

Reply:
`switch (queryContext.contextType) {
      case ContextType.Undefined:
        all = keywordsCompletions;
        break;
      case ContextType.Table:
        all = await sqlAutocompleter.getCompletionItemsFromQueryContext(
          queryContext
        );
        break;
      case ContextType.Column:
        console.log("in case C")
        all = await sqlAutocompleter.getCompletionItemsFromQueryContext(
          queryContext
        );
        const filteredCompletions: object[] = all.filter(obj => obj.value.includes(prefix));
        searchInObject = filteredCompletions.length === 0;
        if (searchInObject) {
          all = await sqlAutocompleter.getCompletionItemsFromQueryContextColumn(
            queryContext
          );
        }
        break;
    }`

Reply:
why does it come out so ugly !?!?